src/socket-client.ts

```diff
@@ -119,7 +118,12 @@ export class SocketClient extends EventEmitter {
 
 
     public async connect(): Promise<any> {
-        const socket = SocketIOClient.connect(this.socketUrl, {
+        const parsedUrl = new URL(this.socketUrl);
+        const path = parsedUrl.pathname && parsedUrl.pathname !== "/" ?
```

Comment:
I would appreciate a multi-line comment block explaining why we do this. If we see this in 2 years, we will easily remove it :D!